src/internal_modules/roc_packet/shipper.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2023 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_packet/shipper.h
+//! @brief Prepare and ship outgoing packets.
+
+#ifndef ROC_PACKET_SHIPPER_H_
+#define ROC_PACKET_SHIPPER_H_
+
+#include "roc_address/socket_addr.h"
+#include "roc_core/noncopyable.h"
+#include "roc_packet/icomposer.h"
+#include "roc_packet/iwriter.h"
+
+namespace roc {
+namespace packet {
+
+//! Prepare a packet for being sent.
+class Shipper : public IWriter, public core::NonCopyable<> {
+public:
+    //! Initialize.
+    //!  - @p address - address that is assigned to a packet.
+    //!  - @p composer to compose a packet if necessary.
+    //!  - @p writer to write outgoing packet.
+    Shipper(const address::SocketAddr& address, IComposer& composer, IWriter& writer);
```

Comment:
nit: better rename address to dst_address or something like this, to avoid confusion, because packet has two addresses: src and dst.